src/consensus/aft/test/driver.h

```diff
@@ -899,6 +919,65 @@ class RaftDriver
     }
   }
 
+  void assert_commit_safety(ccf::NodeId node_id, const size_t lineno)
```

Comment:
Is this `QuorumLoginv` in ccfraft.tla?

https://github.com/microsoft/CCF/blob/75b3476bff4c8f3c35954883c08b53cd64fc02df/tla/ccfraft.tla#L1193-L1197

Reply:
Yes. The violation that `suffix_collision.3` exposed wasn't easily detected by the existing asserts - we could have healed the network and let it try to catch up for a while and then seen that state-sync was never reached, but that's a little distant. So I added this assertion, which aims to match the semantics of `QuorumLogInv`.

Reply:
I'd suggest to add a comment that points at the spec's `QuorumLogInv`.